src/functions/leastSquaresDirect.jl

```diff
@@ -67,7 +76,7 @@ function prox!(y::AbstractVector{C}, f::LeastSquaresDirect{R, C, M, V, F}, x::Ab
     return (f.lambda/2)*norm(f.res, 2)^2
 end
 
-function factor_step!(f::LeastSquaresDirect{R, C, M, V, F}, gamma::R) where {R, C, M <: DenseMatrix, V, F}
+function factor_step!(f::LeastSquaresDirect{R, C, M, V, F}, gamma::R) where {R, C, M, V, F}
```

Comment:
I thought you meant removing the restriction on `F` in line 111 below

Reply:
You are right, I missed that. But I think both are decent fallbacks, and they are both needed if we should have any chance of solving for general types. Added the change in line 111 also